### PR TITLE
Algorithms

### PR DESCRIPTION
**Pull request by KitlerUA**

### CODE REVIEW

File: algo/fight.go [hunk 1]
```diff
@@ -0,0 +1,39 @@
+package algo
+
+import (
+	"github.com/bohdanlisovskyi/Golang-252-Telegram-Bot-Game/database/postgres"
+)
+
+const randomRange = 0.25
```
Comment (YAWAL) [inline algo/fight.go]: Constants should be named in capital letters (RANDOM_RANGE)

File: algo/fleet.go [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import (
+	"errors"
+	"math/rand"
+	"time"
+
+	"github.com/bohdanlisovskyi/Golang-252-Telegram-Bot-Game/database/postgres"
+)
+
+//FleetIsAlive - return true if exists at least one ship with Health > 0, false otherwise
+func FleetIsAlive(fleet []postgres.Ship) bool {
```
Comment (YAWAL) [inline algo/fleet.go]: Better name - IsFleetAlive

File: algo/fleet.go [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+			summaryHit += int(float64(ship.HitRate) * (1.0 + rGen.Float64()*randomRange))
+		} else {
+			summaryHit += int(float64(ship.HitRate) * (1.0 - rGen.Float64()*randomRange))
+		}
+	}
+	return summaryHit
+}
+
+//ComputeFlightTime - return time needed to rich destination between planets
+func ComputeFlightTime(fleet []postgres.Ship, startPlanet, endPlanet *postgres.Planet) (time.Duration, *error) {
```
Comment (YAWAL) [inline algo/fleet.go]: Why it returns pointer *error but not an error?

File: algo/fleet.go [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+	speed, ok := FindMinimalLevel(fleet)
+	if !ok {
+		err := errors.New("Cannot find fleet's speed")
+		return 0, &err
```
Comment (YAWAL) [inline algo/fleet.go]: Maybe, add some log for this error, using our loger.

File: algo/fleet.go (hunk 3 above)
Comment (KitlerUA) [reply algo/fleet.go]: You are right. I will fix it

File: algo/fleet.go (hunk 2 above)
Comment (KitlerUA) [reply algo/fleet.go]: Fixed

**Pull request merged**